test/e2e/validation_test.go

```diff
@@ -24,6 +24,65 @@ import (
 	nc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine/nodeconfig"
 )
 
+// testNodeMetadata tests if all nodes have a worker label and kubelet version and are annotated with the version of
+// the currently deployed WMCO
+func testNodeMetadata(t *testing.T) {
+
+	// The worker label is not actually added by WMCO however we would like to validate if the Machine Api is properly
+	// adding the worker label, if it was specified in the MachineSet. The MachineSet created in the test suite has
+	// the worker label
+	workerLabelTest := "Worker Label"
+	kubeletVersionTest := "Kubelet Version"
+	versionAnnotationTest := "Version Annotation"
+
+	operatorVersion, err := getWMCOVersion()
+	require.NoError(t, err, "could not get WMCO version")
+
+	clusterKubeletVersion, err := getClusterKubeVersion()
+	require.NoError(t, err, "could not get cluster kube version")
+
+	testCases := []struct {
+		testCase string
+		check    string
+	}{
+		{
+			testCase: workerLabelTest,
+			check:    nc.WorkerLabel,
+		},
+		{
+			testCase: kubeletVersionTest,
+			check:    clusterKubeletVersion,
+		},
+		{
+			testCase: versionAnnotationTest,
+			check:    nc.VersionAnnotation,
+		},
+	}
+
+	for _, node := range gc.nodes {
+		for _, test := range testCases {
+			t.Run(node.GetName(), func(t *testing.T) {
+				if test.testCase == kubeletVersionTest {
+					t.Run(test.testCase, func(t *testing.T) {
+						assert.Contains(t, node.Status.NodeInfo.KubeletVersion, test.check, "expected kubelet version was not present on %s", node.GetName())
+					})
+				}
+				if test.testCase == workerLabelTest {
+					t.Run(test.testCase, func(t *testing.T) {
+						assert.Contains(t, node.Labels, test.check, "expected node label %s was not present on %s", nc.WorkerLabel, node.GetName())
+					})
+				}
+				if test.testCase == versionAnnotationTest {
+					t.Run(test.testCase, func(t *testing.T) {
+						require.Containsf(t, node.Annotations, test.check, "node %s missing version annotation", node.GetName())
+						assert.Equal(t, operatorVersion, node.Annotations[test.check], "WMCO version annotation mismatch")
+					})
+				}
+			})
+		}
+	}
```

Comment:
I find this a bit hard to read as they're not really different cases of the same test, but different tests. Not sure how others feel about it.
I'd propose:
```suggestion
	for _, node := range gc.nodes {
		t.Run(node.GetName(), func(t *testing.T) {
			t.Run(kubeletVersionTest, func(t *testing.T) {
				assert.Contains(t, node.Status.NodeInfo.KubeletVersion, clusterKubeletVersion, "expected kubelet version was not present on %s", node.GetName())
			})
			t.Run(workerLabelTest, func(t *testing.T) {
				assert.Contains(t, node.Labels, nc.WorkerLabel, "expected node label %s was not present on %s", nc.WorkerLabel, node.GetName())
			})
			t.Run(versionAnnotationTest, func(t *testing.T) {
				require.Containsf(t, node.Annotations, nc.VersionAnnotation, "node %s missing version annotation", node.GetName())
				assert.Equal(t, operatorVersion, node.Annotations[test.check], "WMCO version annotation mismatch")
			})
		})
	}
```
```

Reply:
agree. Changed